DESIGN.md

```diff
@@ -0,0 +1,247 @@
+# Linux Worker Service using Namespaces/CGroups Implemented in Golang
+
+This is an experiement to build a job worker service that provides an API to run arbitrary Linux processes.
+
+There will be 3 components to make up this service:
+
+1. A Golang library to start, stop, and query the status/output of a worker job
+2. A gRPC API to expose the functionality of this library over a network to start, stop, query status and stream output of a job, including authentication via mTLS and authorization scheme
+3. A CLI for users to interact with the API to start, stop, query status/stream output of a job
+
+> :pencil: Note that code and command samples here are meant to be approximations of the final implementation. Slight variations may exist in what is actually implemented in code from what you see in this design.
+
+## Library
+
+This will be the lowest-level part of the service stack, with the following goals and outcomes:
+
+* providing an interface for a linux command/process to be executed on behalf of a user/owner
+* interfacing with the linux OS internals/kernel via [`sys/unix`](https://pkg.go.dev/golang.org/x/sys/unix) to:
```

Comment:
Can the `os/exec` standard package be used instead of `x/sys/unix`? What functionality is missing from `os/exec` that would require the use of `x/sys/unix` to start run a command  ?